### PR TITLE
Merge upstream

### PR DESCRIPTION
**Pull request by pbrubeck**

### CODE REVIEW

File: .github/workflows/tsfc-tests.yml [hunk 1]
```diff
@@ -28,18 +28,18 @@ jobs:
           pip3 install .
 
       - name: Clone tsfc
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           path: ./tsfc
           repository: firedrakeproject/tsfc
-          ref: master
```
Comment (pbrubeck) [inline .github/workflows/tsfc-tests.yml:35]: @mscroggs How do we revert this change?

File: .github/workflows/tsfc-tests.yml [hunk 2]
```diff
@@ -28,18 +28,18 @@ jobs:
...
+          ref: mscroggs/gdim
       - name: Install tsfc
         run: |
           cd tsfc
           pip install -r requirements-ext.txt
           pip install git+https://github.com/coneoproject/COFFEE.git#egg=coffee
           pip install git+https://github.com/firedrakeproject/fiat.git#egg=fenics-fiat
-          pip install git+https://github.com/FInAT/FInAT.git#egg=finat
```
Comment (pbrubeck) [inline .github/workflows/tsfc-tests.yml:42]: also here

File: .github/workflows/tsfc-tests.yml (hunk 1 above)
Comment (mscroggs) [reply .github/workflows/tsfc-tests.yml:35]: If we merge https://github.com/FEniCS/ufl/pull/326 then update in this fork that should do it I hope

**Review by connorjward — APPROVED** [review]
